.circleci/config.yml

```diff
@@ -0,0 +1,96 @@
+version: 2
+job_defaults: &job_defaults
+    docker:
+        - image: circleci/openjdk:10-jdk-sid-node-browsers-legacy
+    working_directory: ~/scratch
+
+jobs:
+    setup:
+      <<: *job_defaults
+      steps:
+        - checkout
+        - run: mvn dependency:go-offline
+        - save_cache:
+            paths:
+              - ~/.m2
+            key: v1-dependencies-{{ checksum "pom.xml" }}
+
+    validation:
+      <<: *job_defaults
+      steps:
+        - checkout
+        - restore_cache:
+            keys:
+              - v1-dependencies-{{ checksum "pom.xml" }}
+              - v1-dependencies-
+        - run: mvn clean test
+    build:
+      <<: *job_defaults
+      steps:
+        - checkout
+        - restore_cache:
+            keys:
+              - v1-dependencies-{{ checksum "pom.xml" }}
+              - v1-dependencies-
+#        - run: python increment.py VERSION
+        - run: printf "$mvn_setting" > ~/.m2/settings.xml
+        - run: printf -- "$gpg_secret" | gpg --batch --yes --passphrase=$gpg_password --pinentry-mode loopback --import
+#        - run: mvn deploy -Dgpg.passphrase=$gpg_password
```

Comment:
Add a comment above it, saying that we should uncomment if we want to deploy snapshot release